### PR TITLE
Addition of Fuselage Engine Modeling

### PR DESCRIPTION
**Pull request by Dawson-Manning**
### Summary

This commit increases Aviary's (version: 0.9.4-dev) functionality by adding statements to allow of the modeling of aircraft that only utilize fuselage engines. Previously modeling aircraft that only use fuselage mounted engines was not supported. This addition was checked with models that have no wing engines (2 fuselage engines), 2 wing engines (no fuselage engines), and 4 wing engines (no fuselage engines) and all optimized successfully. Before and after edit images attached.
Screenshot 2024-07-26 142041
Screenshot 2024-07-26 142014

### Related Issues

- Resolves Issue #342

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

**Comment by Dawson-Manning** [comment]
Corrections made to fix failure of "test_wing_detailed.py:DetailedWingBendingTest.test_case_multiengine"

File: aviary/subsystems/mass/flops_based/wing_detailed.py [hunk 1]
```diff
@@ -195,9 +199,11 @@ def compute(self, inputs, outputs):
         # i is the counter for which engine model we are checking
         for i in range(num_engine_type):
             # idx2 is the last index for the range of engines of this type
-            idx2 = idx + int(num_wing_engines[i]/2)
-
-            eng_loc = engine_locations[idx:idx2][0]
+            idx2 = idx + int(num_wing_engines[i] / 2)
+            if num_wing_engines[i] > 0:
+                eng_loc = engine_locations[idx:idx2][0]
+            else:
```
Comment (Kenneth-T-Moore) [inline aviary/subsystems/mass/flops_based/wing_detailed.py:205]: I verified with Jason that this is the right behavior.

File: aviary/subsystems/mass/flops_based/wing_detailed.py [hunk 2]
```diff
@@ -212,8 +218,12 @@ def compute(self, inputs, outputs):
 
                 delme = dy * eel[1:]
 
-                delme[loc[-1]] = engine_locations[idx:idx2][0] - \
-                    integration_stations[loc[-1]]
+                if num_wing_engines[i] > 0:
+                    delme[loc[-1]] = engine_locations[idx:idx2][0] - \
+                        integration_stations[loc[-1]]
+                else:
```
Comment (Kenneth-T-Moore) [inline aviary/subsystems/mass/flops_based/wing_detailed.py:224]: Probably don't need the "else" branch anymore, since num_wing_engines=0 bails before it reaches this code now.

**Review by Kenneth-T-Moore — APPROVED** [review]
Looks good.

**Review by jkirk5 — CHANGES_REQUESTED** [review]
Test values need additional investigation (addition of wing engines did not change bending factor for the multiengine test

**Review by jkirk5 — APPROVED** [review]
I resolved the edge case for multiple engines of a given type (multi-engine ended up being fine)
